New Integration Release Cisco Umbrella DNS v1.0.0 (#16503)
DataDog/integrations-core: cisco_umbrella_dns/README.md

# Cisco Umbrella DNS Integration for Datadog

## Overview

> Cisco Umbrella is the leading platform for network DNS security monitoring. Umbrella's DNS-layer security offers a fast and easy way to enhance security, providing improved visibility and protection for users both on and off the network. By preventing threats over any port or protocol before they reach the network or endpoints, Umbrella DNS-layer security aims to deliver the most secure, reliable, and fastest internet experience to over 100 million users.

Reviewer: We usually include a link to the vendor's website. Could you link to their website here too?

Reviewer 2: Done

> This integration offers visualizations and log enrichments for Cisco Umbrella DNS and Proxy logs.

Reviewer 3: little nit here:

Its unclear what log enrichment means. I doubt this is using actual enrichment via a lookup processor. Does this mean theres is a pipeline? or are there dashboards populated by logs?

Should also include, at a high level: 

what are some examples of logs ? 
why would users want to ingest them into the platform? 
what benefits do they get from provided visualizations?

Reviewer 4: +1

Reviewer 2: Updated the README, please have a look.

# Cisco Umbrella DNS Integration for Datadog

## Overview

[Cisco Umbrella][1] is the leading platform for network DNS security monitoring. Umbrella's DNS-layer security offers a fast and easy way to enhance security, providing improved visibility and protection for users both on and off the network. By preventing threats over any port or protocol before they reach the network or endpoints, Umbrella DNS-layer security aims to deliver the most secure, reliable, and fastest internet experience to over 100 million users.

> Cisco Umbrella DNS integration ingests DNS and Proxy logs to Datadog.  Datadog integration log pipeline enriches logs to include geo-graphical information of client and server hosts and normalizes data to datadog standard attributes. This integration offers visualizations and provides detailed insights into DNS traffic and proxied requests, including total DNS requests, allowed/blocked domains, top blocked categories, proxied traffic over time, and more via dashboards.

Reviewer 5: Does this connect to SIEM? If so, should that be mentioned here?

> Cisco Umbrella DNS integration ingests DNS and Proxy logs to Datadog.  Datadog integration log pipeline enriches logs to include geo-graphical information of client and server hosts and normalizes data to datadog standard attributes. This integration offers visualizations and provides detailed insights into DNS traffic and proxied requests, including total DNS requests, allowed/blocked domains, top blocked categories, proxied traffic over time, and more via dashboards.

Reviewer 4: I ended up re-writing it. See below:

The Cisco Umbrella DNS integration collects DNS and Proxy logs and sends them to Datadog. Using the out-of- the-box logs pipeline, the logs are parsed and enriched for easy searching and analysis. This integration includes several dashboards visualizing total DNS requests, allowed/blocked domains, top blocked categories, proxied traffic over time, and more. If you have Datadog Cloud SIEM, Umbrella DNS logs will be analyzed by threat intelligence for matches against common attacker destinations. DNS logs are also useful for threat hunting and during investigations to compliment logs from other sources.

> Cisco Umbrella DNS integration ingests DNS and Proxy logs to Datadog.  Datadog integration log pipeline enriches logs to include geo-graphical information of client and server hosts and normalizes data to datadog standard attributes. This integration offers visualizations and provides detailed insights into DNS traffic and proxied requests, including total DNS requests, allowed/blocked domains, top blocked categories, proxied traffic over time, and more via dashboards.

Reviewer 2: Done

## Setup

### Configuration

#### Cisco Umbrella DNS Configuration

> 1. Login to [**Umbrella** > **Login**][2] with your credentials.
> 2. From the left panel, click on **Admin**.
> 3. Then, click on **API Keys**.
> 4. Create a new API Key using the **Add** button.
> 5. Apply `reports.aggregations:read` and `reports.granularEvents:read` key scopes to the API key.
> 6. Copy the API Key and Key Secret.

Reviewer 5: ```suggestion
1. Login to [**Umbrella**][2] with your credentials.
2. From the left panel, select **Admin**.
3. Select **API Keys**.
4. Create a new API Key.
5. Apply the `reports.aggregations:read` and `reports.granularEvents:read` key scopes to the API key.
6. Copy the API Key and Key Secret, which will be used during the next portion of configuration steps.
```

Reviewer 2: Done

#### Cisco Umbrella DNS DataDog Integration Configuration

Configure the Datadog endpoint to forward Cisco Umbrella DNS events as logs to Datadog.

1. Navigate to `Cisco Umbrella DNS`.

> 2. Provide your Cisco Umbrella DNS credentials and click on `Add`.

Reviewer 5: ```suggestion
2. Add your Cisco Umbrella DNS credentials.
```

Reviewer 2: Done

# Cisco Umbrella DNS Integration for Datadog

## Overview

Cisco Umbrella is the leading platform for network DNS security monitoring. Umbrella's DNS-layer security offers a fast and easy way to enhance security, providing improved visibility and protection for users both on and off the network. By preventing threats over any port or protocol before they reach the network or endpoints, Umbrella DNS-layer security aims to deliver the most secure, reliable, and fastest internet experience to over 100 million users.

This integration offers visualizations and log enrichments for Cisco Umbrella DNS and Proxy logs.

> ### Supported Logs:

Reviewer 6: ```suggestion
### Supported Logs
```

Reviewer 2: Done

- Cisco Umbrella DNS Logs
- Cisco Umbrella Proxy Logs

## Setup

### Configuration

#### Cisco Umbrella DNS Configuration

> 1. Login via [Umbrella > Login](https://login.umbrella.com/) with your credentials
> 2. From the left panel, click on Admin
> 3. Then, click on API Keys
> 4. Create a new API Key using Add button
> 5. Apply reports.aggregations:read and reports.granularEvents:read key scopes to the API key
> 6. Copy the API Key and Key Secret

Reviewer 6: ```suggestion
1. Login to [**Umbrella** > **Login**][1] with your credentials.
2. From the left panel, click on **Admin**.
3. Then, click on **API Keys**.
4. Create a new API Key using the **Add** button.
5. Apply `reports.aggregations:read` and `reports.granularEvents:read` key scopes to the API key.
6. Copy the API Key and Key Secret.
```

Reviewer 2: Done

#### Cisco Umbrella DNS DataDog Integration Configuration

Configure the Datadog endpoint to forward Cisco Umbrella DNS events as logs to Datadog.

1. Navigate to `Cisco Umbrella DNS`.
2. Provide your Cisco Umbrella DNS credentials and click on `Add`.

| Cisco Umbrella DNS Parameters | Description                                                                |
| ----------------------------- | -------------------------------------------------------------------------- |
| Host                          | The API server host of Cisco Umbrella. Default host is `api.umbrella.com`. |

> | API Key                       | The API Key from Cisco Umbrella                                            |
> | Key Secret                    | The Key Secret from Cisco Umbrella                                         |

Reviewer 6: ```suggestion
| API Key                       | The API Key from Cisco Umbrella.                                            |
| Key Secret                    | The Key Secret from Cisco Umbrella.                                         |
```

Reviewer 2: Done

## Data Collected

### Logs

The integration collects and forwards Cisco Umbrella DNS and Proxy logs to Datadog.

### Metrics

The Cisco Umbrella DNS integration does not include any metrics.

### Events

The Cisco Umbrella DNS integration does not include any events.

## Support

> For further assistance, contact [Datadog Support](https://docs.datadoghq.com/help/).

Reviewer 6: ```suggestion
For further assistance, contact [Datadog Support][2].

[1]: https://login.umbrella.com/
[2]: https://docs.datadoghq.com/help/
```

Reviewer 2: Done
